fnmatch/fnmatch.py

```diff
@@ -99,6 +137,11 @@ def translate(pat):
                     stuff = '\\' + stuff
                 res = '%s[%s]' % (res, stuff)
         else:
+         if (sys.platform == 'uefi') and (sys.implementation.name == 'micropython') :
```

Comment:
Probably niver t make this a 'constant' at the top of the file. However, is this check always correct?

Reply:
This code is correct; all though; there seems to be some indentation issues. 
I'll re-check on my master file. That said the "else" goes with the 
`elif stuff[0] == '^':`
above it so that the _re_escape(c) is called in the case of uefi implementation.